### PR TITLE
fix: Add autodevops template.

### PR DESCRIPTION
**Pull request by gary-van-woerkens**

### CODE REVIEW

File: templates/autodevops/components/app.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        container: {
+          resources: resources ?? {
+            limits: {
+              cpu: "200m",
+              memory: "256Mi",
+            },
+            requests: {
+              cpu: "50m",
+              memory: "128Mi",
+            },
```
Comment (revolunet) [inline templates/autodevops/components/app.ts:74]: on pourrait mettre ca dans le `createDeployment` par défaut ? https://github.com/SocialGouv/kosko-charts/blob/master/src/utils/createDeployment.ts#L117

File: templates/autodevops/components/pg.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  const { azurepg } = await Config();
+
+  if (!azurepg) {
+    return [];
+  }
+
+  if (env.env === "dev") {
+    return create("pg-user", { env });
+  }
+
```
Comment (revolunet) [inline templates/autodevops/components/pg.ts:22]: a priori pas besoin de ce qui est en-dessous : https://github.com/SocialGouv/kosko-charts/releases/tag/v9.0.0

File: templates/autodevops/components/yaml.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import env from "@kosko/env";
+import { importYamlFolder } from "@socialgouv/kosko-charts/components/yaml";
+import path from "path";
+
+const manifests = importYamlFolder(
+  path.join(__dirname, "..", `environments/${env.env}/yaml`)
+);
+
```
Comment (revolunet) [inline templates/autodevops/components/yaml.ts:8]: a priori après avoir importé les YAML, il faut leur appliquer les metadata : https://github.com/SocialGouv/carnet-de-bord/blob/eee524a268b180d66744ce7babcf49bbde21a251/.k8s/components/yaml.ts#L12

File: templates/autodevops/kosko.toml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+components = ["*"]
```
Comment (revolunet) [inline templates/autodevops/kosko.toml:1]:
```suggestion
components = ["*", "!jobs"]
```

File: templates/autodevops/utils/addIngressAnnotations.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  /* pass dynamic deployment URL as env var to the container */
+  if (!ingress.metadata) {
+    ingress.metadata = { annotations: {} };
+  }
+  ingress.metadata.annotations = {
+    ...(ingress.metadata.annotations ?? {}),
+    ...annotations,
+  };
+  return ingress;
+};
```
Comment (revolunet) [inline templates/autodevops/utils/addIngressAnnotations.ts:16]: could be moved to `src/utils` ?

File: templates/autodevops/components/app.ts [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+          {}
+        )
+      : {}),
+    ...configProbes,
+  };
+
+  if (type && type === "app") {
+    const manifests = await create(name, {
+      config: {
+        containerPort: 3000,
```
Comment (revolunet) [inline templates/autodevops/components/app.ts:60]:
```suggestion
```

File: templates/autodevops/components/app.ts [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+          resources: resources ?? {
+            limits: {
+              cpu: "200m",
+              memory: "256Mi",
+            },
+            requests: {
+              cpu: "50m",
+              memory: "128Mi",
+            },
+          },
```
Comment (revolunet) [inline templates/autodevops/components/app.ts:75]:
```suggestion
          resources,
```

File: e2e/templates/autodevops/app/__tests__/__snapshots__/kosko generate --env dev.ts.snap [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+      labels:
+        app: myapp
+        application: e2e-branch-42-sample-kosko
+        owner: sample-kosko
+        team: sample-kosko
+        cert: wildcard
+    spec:
+      containers:
+        - image: >-
+            harbor.fabrique.social.gouv.fr/undefined/myapp:8843083edb7f873cad1d1420731a60773594ffae
```
Comment (revolunet) [inline e2e/templates/autodevops/app/__tests__/__snapshots__/kosko generate --env dev.ts.snap:63]: missing HARBOR_PROJECT ?

File: e2e/templates/autodevops/app/__tests__/__snapshots__/kosko generate --env prod.ts.snap [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+              service:
+                name: myapp
+                port:
+                  name: http
+            path: /
+            pathType: Prefix
+  tls:
+    - hosts:
+        - e2e-branch-42-sample-kosko.dev2.fabrique.social.gouv.fr
+      secretName: wildcard-crt"
```
Comment (revolunet) [inline e2e/templates/autodevops/app/__tests__/__snapshots__/kosko generate --env prod.ts.snap:161]: looks like a dev certificate and hostname

File: e2e/templates/autodevops/app/__tests__/__snapshots__/kosko generate --env prod.ts.snap [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+      labels:
+        app: myapp
+        application: e2e-branch-42-sample-kosko
+        owner: sample-kosko
+        team: sample-kosko
+        cert: wildcard
+    spec:
+      containers:
+        - image: >-
+            harbor.fabrique.social.gouv.fr/undefined/myapp:8843083edb7f873cad1d1420731a60773594ffae
```
Comment (revolunet) [inline e2e/templates/autodevops/app/__tests__/__snapshots__/kosko generate --env prod.ts.snap:63]: we should have tagged version here

File: e2e/templates/autodevops/app/__tests__/__snapshots__/kosko generate --env preprod.ts.snap [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+      labels:
+        app: myapp
+        application: e2e-branch-42-sample-kosko
+        owner: sample-kosko
+        team: sample-kosko
+        cert: wildcard
+    spec:
+      containers:
+        - image: >-
+            harbor.fabrique.social.gouv.fr/undefined/myapp:8843083edb7f873cad1d1420731a60773594ffae
```
Comment (revolunet) [inline e2e/templates/autodevops/app/__tests__/__snapshots__/kosko generate --env preprod.ts.snap:63]: we should have tag here

**Comment by gary-van-woerkens** [comment]
We also have to manage Rancher IDs within the namespace definition.

**Pull request closed without merging**